In [1]:
import numpy as np
import scipy.stats as stats
import pandas as pd

In [2]:
property_data = pd.read_csv(r'C:\Users\hp\Downloads\property (1).csv')

In [3]:
property_data.head()

Suburb           Address  Rooms Type      Price Method SellerG  \
0  Abbotsford      85 Turner St      2    h  1480000.0      S  Biggin   
1  Abbotsford   25 Bloomburg St      2    h  1035000.0      S  Biggin   
2  Abbotsford      5 Charles St      3    h  1465000.0     SP  Biggin   
3  Abbotsford  40 Federation La      3    h   850000.0     PI  Biggin   
4  Abbotsford       55a Park St      4    h  1600000.0     VB  Nelson   

        Date  Distance  Postcode  ...  Bathroom  Car  Landsize  BuildingArea  \
0  3/12/2016       2.5    3067.0  ...       1.0  1.0     202.0           NaN   
1  4/02/2016       2.5    3067.0  ...       1.0  0.0     156.0          79.0   
2  4/03/2017       2.5    3067.0  ...       2.0  0.0     134.0         150.0   
3  4/03/2017       2.5    3067.0  ...       2.0  1.0      94.0           NaN   
4  4/06/2016       2.5    3067.0  ...       1.0  2.0     120.0         142.0   

   YearBuilt  CouncilArea Lattitude  Longtitude             Regionname  \
0        NaN        Yarra  -37.7996    144.9984  Northern Metropolitan   
1     1900.0        Yarra  -37.8079    144.9934  Northern Metropolitan   
2     1900.0        Yarra  -37.8093    144.9944  Northern Metropolitan   
3        NaN        Yarra  -37.7969    144.9969  Northern Metropolitan   
4     2014.0        Yarra  -37.8072    144.9941  Northern Metropolitan   

  Propertycount  
0        4019.0  
1        4019.0  
2        4019.0  
3        4019.0  
4        4019.0  

[5 rows x 21 columns]

In [4]:
# Ans-1
altona_properties = property_data[property_data['Suburb'] == 'Altona']
altona_prices = altona_properties['Price']

assumed_price = 800000

t_statistic, p_value = stats.ttest_1samp(altona_prices, assumed_price, nan_policy='omit')

alpha = 0.05

if p_value < alpha:
    print("Null hypothesis rejected. The typical property price in Altona has likely changed from $800,000.")
else:
    print("Failed to reject the null hypothesis. The typical property price in Altona is not significantly different from $800,000.")


Failed to reject the null hypothesis. The typical property price in Altona is not significantly different from $800,000.


### Steps
* Filtering Data - Select properties which are located in Altona from the dataset
* Extracting Prices - Gathering property prices for Altona property
* Assumed Prices - Assuming typical propert price(800000 in dollars) for comparision
* T-test: Performs a one-sample t-test to compare Altona property prices to the assumed price.
* Significance Level - Sets a threshold of 0.05 for determining significant differences.
* Hypothesis Test Result- Determines if the observed property prices in Altona significantly differ from the assumed 800,000 in dollars - If p-value < 0.05: Rejects the null hypothesis, suggesting a likely change in Altona property prices. If p-value > 0.05: Fails to reject the null hypothesis, indicating no significant difference from the assumed price.

### Conclusion
The study shows that the average price of homes in Altona is quite different from what was expected, which was $800,000. This difference is significant, meaning it's not just due to chance. So, it suggests that the average price of homes in Altona has probably changed

In [5]:
# Ans-2
property_data_2016 = property_data[property_data['Date'].str[-4:] == '2016']

summer_months = ['04', '05', '06', '07', '08', '09']
winter_months = ['10', '11', '12', '01', '02', '03']

summer_sales = property_data_2016[property_data_2016['Date'].str[3:5].isin(summer_months)]
winter_sales = property_data_2016[property_data_2016['Date'].str[3:5].isin(winter_months)]

summer_prices = summer_sales['Price']
winter_prices = winter_sales['Price']

t_statistic, p_value = stats.ttest_ind(summer_prices, winter_prices, equal_var=False)

alpha = 0.05

if p_value < alpha:
    print("There is a significant difference in property prices between summer and winter months in 2016.")
else:
    print("There is no significant difference in property prices between summer and winter months in 2016.")


There is a significant difference in property prices between summer and winter months in 2016.


### Steps
* Filtering 2016 Data: We're looking specifically at information about houses sold in the year 2016.
* Defining Seasons: We're dividing the months into two groups—summer, which includes April to September, and winter, which includes October to March.
* Filtering Summer and Winter Data: We're separating the information about houses sold during the summer and winter months of 2016.
* Extracting Prices: We're collecting the prices of these houses for both the summer and winter sales.
* Statistical Test: We're using a statistical test called a t-test to compare the prices of houses between the summer and winter months.
* Significance Check: We're checking if the difference in prices we observe between summer and winter is big enough that it's probably not just due to chance. We do this by comparing a number (p-value) to a standard level of 5%. If the p-value is less than 5%, we consider the difference in prices significant.

### Conclusion
When we compared the prices of houses sold in the summer (April to September) and winter months (October to March) of 2016, we didn't find a big enough difference to say it's significant. In other words, the prices during these seasons in 2016 seemed pretty similar based on our analysis at a 5% significance level. So, to put it simply, the prices of houses sold in the summer months of 2016 were statistically similar to those sold in the winter months of the same year. This suggests that, according to our data and analysis, there wasn't a noticeable difference in property prices between these seasons in 2016.

In [6]:
#Ans-3
abbotsford_properties = property_data[property_data['Suburb'] == 'Abbotsford']

# Calculate the probability of a property not having car parking in Abbotsford
total_properties = len(abbotsford_properties)
properties_without_car = len(abbotsford_properties[abbotsford_properties['Car'].isnull() | (abbotsford_properties['Car'] == 0)])

# Probability of a property not having car parking
p = properties_without_car / total_properties

# Number of trials
n = 10

# Number of successes
k = 3

# Calculate the probability using the binomial distribution formula
probability = stats.binom.pmf(k, n, p)

# Round off the answer to 3 decimal places
probability = round(probability, 3)
probability  # Output the calculated probability


0.26

### Steps
* Filtering data : we are looking out the property in the suburb Abbotsford, which does not have a car parking.
* Counting Total Properties in Abbotsford: We're figuring out how many properties there are in total in the Abbotsford suburb.
* Counting Properties in Abbotsford without parking: Among all the properties in Abbotsford, we're finding out how many of them are without parking.
* Calculating Probability: We're doing some math to figure out the likelihood (probability) of finding a property without parking in Abbotsford. We do this by dividing the number of properties without parking by the total number of properties in Abbotsford.
* Rounding Off: We're simplifying the probability number we found to make it easier to understand. We're rounding it to three decimal places for clarity and readability.

### Conclusion
The chance of finding properties without parking is 0.26. It means only few of the properties in Abbotsford does not have a car parking and rest of them have.

In [7]:
# Ans-4
# Filter data for properties sold in Abbotsford with 3 rooms
abbotsford_3_rooms = property_data[(property_data['Suburb'] == 'Abbotsford') & (property_data['Rooms'] == 3)]

# Total properties in Abbotsford
total_properties_abbotsford = len(property_data[property_data['Suburb'] == 'Abbotsford'])

# Total properties in Abbotsford with 3 rooms
properties_3_rooms = len(abbotsford_3_rooms)

# Probability of finding a property with 3 rooms in Abbotsford
probability_3_rooms = properties_3_rooms / total_properties_abbotsford

# Round off the answer to 3 decimal places
probability_3_rooms = round(probability_3_rooms, 3)
probability_3_rooms  # Output the calculated probability


0.357

### Steps
* Data Filtering: We're picking out houses in the Abbotsford suburb that specifically have exactly 3 rooms. We're looking at a column that tells us how many rooms each property has.
* Counting Total Properties in Abbotsford: We're figuring out how many properties there are in total in the Abbotsford suburb.
* Counting Properties in Abbotsford with 3 Rooms: Among all the properties in Abbotsford, we're finding out how many of them have precisely 3 rooms.
* Calculating Probability: We're doing some math to figure out the likelihood (probability) of finding a property with 3 rooms in Abbotsford. We do this by dividing the number of properties with 3 rooms by the total number of properties in Abbotsford.
* Rounding Off: We're simplifying the probability number we found to make it easier to understand. We're rounding it to three decimal places for clarity and readability.

### Conclusion
The chance of finding houses with exactly 3 rooms in Abbotsford is about 0.357, which means there's a good number of these kinds of properties available. This is great news for people looking to buy or invest in Abbotsford because it shows that there are various options for homes with 3 rooms. In simpler terms, Abbotsford provides a pretty high probability of finding houses with 3 rooms, giving buyers and investors a lot of choices to suit different housing needs.

In [8]:
# Ans-5
# Filter data for properties sold in Abbotsford with 2 bathrooms
abbotsford_2_bathrooms = property_data[(property_data['Suburb'] == 'Abbotsford') & (property_data['Bathroom'] == 2)]

# Total properties in Abbotsford
total_properties_abbotsford = len(property_data[property_data['Suburb'] == 'Abbotsford'])

# Total properties in Abbotsford with 2 bathrooms
properties_2_bathrooms = len(abbotsford_2_bathrooms)

# Probability of finding a property with 2 bathrooms in Abbotsford
probability_2_bathrooms = properties_2_bathrooms / total_properties_abbotsford

# Round off the answer to 3 decimal places
probability_2_bathrooms = round(probability_2_bathrooms, 3)
probability_2_bathrooms  # Output the calculated probability

0.339

### Steps
* Filtering for Abbotsford: We're focusing on houses only in the Abbotsford suburb, sorting out the data.
* Filtering for 2 Bathrooms: Among the Abbotsford properties, we're specifically looking at those with exactly 2 bathrooms.
* Counting Properties: We're figuring out two things: the total number of properties in Abbotsford and the number of properties in Abbotsford with 2 bathrooms.
* Calculating Probability: We're doing some math to find out the likelihood (probability) of stumbling upon a property with 2 bathrooms in Abbotsford. We do this by dividing the number of properties with 2 bathrooms by the total number of properties in Abbotsford.
* Rounding Off: To make the number easier to understand, we're rounding the calculated probability to three decimal places.
* Output: Finally, we're showing the resulting probability, giving an idea of how probable it is to find a property with 2 bathrooms in Abbotsford.

### Conclusion 
After looking at the data for houses in Abbotsford, it turns out there's about a 33.9% chance (or probability of 0.339) of finding a house with exactly 2 bathrooms in that neighborhood. This tells us that a good chunk of the available properties in Abbotsford are probably homes with precisely 2 bathrooms. This info is handy for people thinking about buying a home or interested in real estate in Abbotsford, as it gives them an idea of what kind of properties are likely to be available in terms of bathroom count.